### PR TITLE
[GPU] Force sdpa use onednn path for prefill and cl path for generation.

### PR DESCRIPTION
**Pull request by ceciliapeng2011**
### Details:
 - *[GPU] Force SDPA use oneDNN path for prefill and clDNN path for generation on ARL-H platform*
 - *...*

### Tickets:
 - *CVS-158461*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp [hunk 1]
```diff
@@ -317,6 +348,17 @@ struct scaled_dot_product_attention_impl : multi_stage_primitive<scaled_dot_prod
             kernels_data.push_back(kernel_selector.get_best_kernel(indirect_kernel_params));
         }
 
+        if (impl_param.get_program().get_engine().get_device_info().gfx_ver.major == 12) { // ARL only
+            sdpa_kernel_params.should_use_sdpa_opt = true;
+            kernels_data.push_back(kernel_selector.get_best_kernel(sdpa_kernel_params));
+        }
+
+        std::cout << "=========== impl_param.get_program().get_engine().get_device_info().gfx_ver=" <<
```
Comment (ceciliapeng2011) [inline src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp:356]: I will check what number is it with ARL. Will update this part later.

File: src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp [hunk 2]
```diff
@@ -233,7 +270,7 @@ struct scaled_dot_product_attention_impl : multi_stage_primitive<scaled_dot_prod
     }
 
 public:
-    static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param, bool is_dynamic, bool indirect = false) {
+    static kernel_params_t get_kernel_params(const kernel_impl_params& impl_param, bool is_dynamic, bool indirect = false, bool force_sdpa_opt = false) {
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp:273]: It seems `force_sdpa_opt` can be removed from this function, as it's being manually set in the create() function

**Review by sshlyapn — APPROVED** [review]
Looks good to me, left minor comments

**Review by p-durandin — APPROVED** [review]

**Review by yeonbok — APPROVED** [review]

**Comment by yeonbok** [comment]
@ceciliapeng2011 LGTM, but is this only effective on ARLH? How about LNL?

**Pull request merged**